src/masternodes/poolpairs.cpp

```diff
@@ -429,6 +429,11 @@ Res CPoolPair::Swap(CTokenAmount in, PoolPrice const & maxPrice, std::function<R
     }
 
     CAmount result = slopeSwap(in.nValue, reserveF, reserveT, height >= Params().GetConsensus().BayfrontGardensHeight);
+    if (height >= Params().GetConsensus().FortCanningHillHeight && result > 0) {
```

Comment:
This should be done at `slopeSwap()` and not outside of it - which then creates inconsistencies on the outcome.